File: Problemas/cpp/bancos-con-herencia/java/Cuenta.java
public abstract class Cuenta
{
	// Comun
	protected double capital;

	public Cuenta(double _capital)
	{
		capital = _capital;
	};

	// Cada tipo de cuenta debe dar su implementación
	public abstract double valor(int t);

	// La clase base ofrece ya una implementación para todos
	public double tae()
	{
		return 100*(valor(12)/valor(0) - 1);
	}

	// La clase base ofrece ya una implementación, que sirve
	// para la mayoria de los casos
	// Los casos especiales pueden redefinirla
	public void actualiza()
	{
		capital = valor(1);
	}
}

File: Problemas/cpp/bancos-con-herencia/c++/cuenta.cc
#include "cuenta.h"
#include <cmath>

using namespace std;

Cuenta::Cuenta(double _capital) : capital(_capital)
{
}

double Cuenta::tae() const
{
	return 100.0*(valor(12)/valor(0) - 1);
}

void Cuenta::actualiza()
{
	capital = valor(1);
}
